tendermint/tests/lite.rs

```diff
@@ -121,138 +120,140 @@ impl MockRequester {
+    ];
+    for dir in &dirs {
+        run_single_step_tests(dir);
+    }
 }
 
-type Trusted = lite::TrustedState<SignedHeader, Header>;
+async fn run_bisection_tests(dir: &str) {
+    let paths = fs::read_dir(PathBuf::from(TEST_FILES_PATH).join(dir)).unwrap();
 
-fn run_test_cases(cases: TestCases) {
-    for (_, tc) in cases.test_cases.iter().enumerate() {
-        let trusted_next_vals = tc.initial.clone().next_validator_set;
-        let mut latest_trusted =
-            Trusted::new(tc.initial.signed_header.clone().into(), trusted_next_vals);
-
-        let expects_err = match &tc.expected_output {
-            Some(eo) => eo.eq("error"),
-            None => false,
-        };
-
-        let trusting_period: std::time::Duration = tc.initial.clone().trusting_period.into();
-        let tm_now = tc.initial.now;
-        let now = tm_now.to_system_time().unwrap();
-
-        for (i, input) in tc.input.iter().enumerate() {
-            println!("i: {}, {}", i, tc.description);
-
-            let untrusted_signed_header = &input.signed_header;
-            let untrusted_vals = &input.validator_set;
-            let untrusted_next_vals = &input.next_validator_set;
-
-            match lite::verify_single(
-                latest_trusted.clone(),
-                &untrusted_signed_header.into(),
-                untrusted_vals,
-                untrusted_next_vals,
-                TrustThresholdFraction::default(),
-                trusting_period,
-                now,
-            ) {
-                Ok(new_state) => {
-                    let expected_state = TrustedState::new(
-                        untrusted_signed_header.clone().into(),
-                        untrusted_next_vals.clone(),
-                    );
-
-                    assert_eq!(new_state, expected_state);
-                    assert!(!expects_err);
-
-                    latest_trusted = new_state.clone();
-                }
-                Err(_) => {
-                    assert!(expects_err);
-                }
-            }
-        }
+    for file_path in paths {
+        let dir_entry = file_path.unwrap();
+        let fp_str = format!("{}", dir_entry.path().display());
+        println!(
+            "Running light client against bisection test-file:\n {:?}",
+            fp_str
+        );
+        let case: TestBisection = read_bisection_test_case(&fp_str);
+        run_bisection_test(case).await;
     }
 }
 
-// Link to the commit where the happy_path.json was created:
-// https://github.com/Shivani912/tendermint/commit/f8305c9a0e05696340fd8853ed5657a2075df895
-#[tokio::test]
-async fn bisection_happy_path() {
-    let case: TestBisection =
-        serde_json::from_str(&read_json_fixture("many_header_bisection/happy_path")).unwrap();
-    run_bisection_test(case).await;
-}
+fn run_single_step_tests(dir: &str) {
+    // TODO: this test need further investigation:
+    let skipped = ["commit/one_third_vals_don't_sign.json"];
```

Comment:
I started with the 4 mentioned tests in that array (from #267). Only that one is failing. So this doesn't really need to be an array now 🤔

Reply:
The 4 tests mentioned are supposed to fail on parsing because they have Absent CommitSigs without validator address. @greg-szabo Did anything change on serialization side here?

Reply:
Oh, just realized the serialization code for CommitSig from Greg hasn't been merged with master yet. I'm not sure if we should compare this branch with the one that has Greg's serialization code or merge the changes with master as is?

Reply:
I'd merge into master. So if Greg finishes the CommitSig changes, we can merge in master into his branch and see if there is any additional work to do to make now failing tests pass.